### PR TITLE
Add Dev Container

### PR DESCRIPTION
**Pull request by getjump**
Dev Container open up a whole new world of possibilities:

1. It is easy to make dev environment even for frontenders, for whom currently it is hard as stated by #178 
2. You can develop even on microwave by utilizing cloud super powers of [Github Codespaces](https://github.com/features/codespaces)

There are also task and launch target defined for VS Code, so that environment is kinda complete, you just start vs code open dev container from source, click run and boom, everything just works (including debugging, and lint and stuff 🔥)

We can probably continue to improve that environment in the future (like for example when we can use Python 3.10 #221). 

Before merge:
- [x] update README
- [x] fix `make help` error
- [x] put python deps into image to speedup container starting
- [x] fix `.gitignore` (ignore whole `./vscode` exclude required for devcontainer)
- [ ] https://github.com/vldc-hq/vldc-bot/issues/237

### CODE REVIEW

**Comment by getjump** [comment]
We probably should store our dependencies in image, since they are not updating that frequently to speed up codespaces/local image builds.

**Review by C-Pro — APPROVED** [review]
👍

**Comment by egregors** [comment]
Looks cool! I'll check this out next week

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.209.6/containers/python-3/.devcontainer/base.Dockerfile
+
+# [Choice] Python version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.10, 3.9, 3.8, 3.7, 3.6, 3-bullseye, 3.10-bullseye, 3.9-bullseye, 3.8-bullseye, 3.7-bullseye, 3.6-bullseye, 3-buster, 3.10-buster, 3.9-buster, 3.8-buster, 3.7-buster, 3.6-buster
+ARG VARIANT="3.9"
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+RUN apt-get -y update && apt-get install -y ffmpeg
+
+# [Optional] If your pip requirements rarely change, uncomment this section to add them to the image.
```
Comment (egregors) [inline .devcontainer/Dockerfile:9]: Looks like include deps into image is a good idea. We change them not so often.

File: .gitignore [hunk 2]
```diff
@@ -103,8 +103,6 @@ venv.bak/
 # mypy
 .mypy_cache/
 
-#vscode
```
Comment (egregors) [inline .gitignore:106]: I believe we can exclude only `launch.json` and `tasks.json` from `.gitignoge`. Personal setting should be ignored.

**Review by egregors — CHANGES_REQUESTED** [review]
Looks pretty cool, really. Especially debugging-in-docker and run in one click. Here are a few improvements proposals and we should provide an updated README for frontends.

**Comment by getjump** [comment]
@egregors I disagree that we should exclude `.vscode` from repository, IT SHOULD NOT contain personal preferences, it should contain project wise preference see https://github.com/microsoft/vscode/tree/main/.vscode (especially https://github.com/microsoft/vscode/blob/main/.vscode/settings.json) for example.

**Comment by C-Pro** [comment]
> @egregors I disagree that we should exclude `.vscode` from repository, IT SHOULD NOT contain personal preferences, it should contain project wise preference

Mine does ¯\_(ツ)_/¯

**Comment by getjump** [comment]
> > @egregors I disagree that we should exclude `.vscode` from repository, IT SHOULD NOT contain personal preferences, it should contain project wise preference
> 
> Mine does ¯_(ツ)_/¯

Can you provide example what's in your project settings?

**Comment by C-Pro** [comment]
> > > @egregors I disagree that we should exclude `.vscode` from repository, IT SHOULD NOT contain personal preferences, it should contain project wise preference
> > 
> > 
> > Mine does ¯_(ツ)_/¯
> 
> Can you provide example what's in your project settings?

`"python.pythonPath": "/home/cpro/.pyenv/versions/3.6.6/bin/python",`

**Comment by egregors** [comment]
> > > > @egregors I disagree that we should exclude `.vscode` from repository, IT SHOULD NOT contain personal preferences, it should contain project wise preference
> > > 
> > > 
> > > Mine does ¯_(ツ)_/¯
> > 
> > 
> > Can you provide example what's in your project settings?
> 
> `"python.pythonPath": "/home/cpro/.pyenv/versions/3.6.6/bin/python",`

Fair enough. Mine contains pyPath as well. ://

**Comment by getjump** [comment]
@egregors @C-Pro Well, I don't like absolute paths relative to local system, probably something like pipenv/venv/etc would be better. But there's not much we can do now. Maybe later we can focus on enhancing process of development without Dev Container, but for now I have found a solution to use settings json specific to container - https://github.com/microsoft/vscode-dev-containers/blob/main/containers/go/.devcontainer/devcontainer.json#L17 . So we can safely put `.vscode/settings.json` into `.gitignore`

**Comment by C-Pro** [comment]
BTW I did not set this path manually. It is changed everytime I select interpreter in vscode via dropdown menu on the bottom.

**Comment by getjump** [comment]
@C-Pro Ye, that is probably not a best place to store interpreter by vscode 🤷‍♂️

**Comment by getjump** [comment]
> - [ ] try to use `docker-compose-dev.yml` for devcontainer

@egregors it seems like __no fix__ to me currently. We should probably somehow remove last line, where we start bot from dev `Dockerfile` or something else.

**Comment by egregors** [comment]
> > * [ ]  try to use `docker-compose-dev.yml` for devcontainer
> 
> @egregors it seems like **no fix** to me currently. We should probably somehow remove last line, where we start bot from dev `Dockerfile` or something else.

Sure, let's talk about it _after NY_ :D :D I believe we could extract it subtask into a separate issue. 

Anyway, your fixes looks grate! I gonna test it locally, just to be sure, and will merge it.

**Comment by getjump** [comment]
@egregors there are a bit work still left, i accidentally commited junk comments, so probably we should merge after NY🌚🌚

**Review by egregors — APPROVED** [review]

**Comment by egregors** [comment]
> @egregors there are a bit work still left, i accidentally commited junk comments, so probably we should merge after NY🌚🌚

@getjump If you are talking about junk in Dockerfile, I cleaned it already. But if smth else here, and you wanna freeze merge for a time, it's okey.

**Comment by getjump** [comment]
@egregors Ye that's probably all. I also added new extension by Microsoft, but we can add it later maybe `"visualstudioexptteam.intellicode-api-usage-examples"`

**Comment by egregors** [comment]
> @egregors Ye that's probably all. I also added new extension by Microsoft, but we can add it later maybe `"visualstudioexptteam.intellicode-api-usage-examples"`

So, in this way I gonna merge it now. @getjump could you please drop a bit more details in https://github.com/vldc-hq/vldc-bot/issues/237?

**Pull request merged**